### PR TITLE
[WIP] rust: update to 1.31.0.

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

**Comment by Cogitri** [comment]
The release is still so fresh that upstream hasn't updated the distfiles yet :)

Since this stabilizes most parts of Rust 2018 we should check if we can update some packages which have needed beta for that so far.

Will ping once fully updated.

**Comment by Gottox** [comment]
https://alpha.de.repo.voidlinux.org/distfiles/rustc-1.31.0-x86_64-unknown-linux-musl.tar.xz
https://alpha.de.repo.voidlinux.org/distfiles/rust-std-1.31.0-x86_64-unknown-linux-musl.tar.xz
currently waiting till cargo finishes.

**Comment by Cogitri** [comment]
Thanks :)

Already compiled on x86_64 and i686. Will test cross to aarch64&armv{6,7}l

**Comment by Gottox** [comment]
cargo is finished too.

**Comment by Cogitri** [comment]
Thanks for the ping. I've tested all glibc platforms and it works so far, will test musl over night.

**Comment by Cogitri** [comment]
Cross compiling seems to work too, musl on arm* and aarch* is the only thing left for testing, I suppose.

File: srcpkgs/cargo/template [hunk 1]
```diff
@@ -10,8 +10,7 @@ maintainer="Enno Boland <gottox@voidlinux.eu>"
 license="MIT, Apache-2.0"
 homepage="https://crates.io/"
 distfiles="https://github.com/rust-lang/cargo/archive/${version}.tar.gz"
-checksum=d14462641610a56fd444200b02bb2547b663de3e63548990fba2e05435f4c096
-_cargo_dist_version=0.30.0
```
Comment (Cogitri) [inline srcpkgs/cargo/template]: I've removed `_cargo_dist_version` here, is there ever a case where we want an older cargo for bootstrapping?

File: srcpkgs/rust/template [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # Template file for 'rust'
 pkgname=rust
-version=1.30.1
+version=1.31.0
 revision=1
-_rust_dist_version=1.30.1
-_cargo_dist_version=0.31.0
+_rust_dist_version=1.31.0
```
Comment (Cogitri) [inline srcpkgs/rust/template]: I suppose we could replace `_rust_dist_version` with `version` too

File: srcpkgs/cargo/template (hunk 1 above)
Comment (Cogitri) [reply srcpkgs/cargo/template]: Ah, nevermind, we want that for when you build the initial musl tooling

File: srcpkgs/rust/template (hunk 2 above)
Comment (Cogitri) [reply srcpkgs/rust/template]: Ah, nevermind, we want that for when you build the initial musl tooling

**Comment by Cogitri** [comment]
As far as I can see this works on all platforms, but please do test this. Excited for Rust 2018 to finally be stabilized and be in Void repos :)

**Comment by jnbr** [comment]
lgtm

**Comment by jnbr** [comment]
merged
3965e0b37088eda04ee75868a3d8e9dcdd800224

**Comment by Cogitri** [comment]
Nice, thanks! :)

**Comment by Cogitri** [comment]
> Anyway I can't find clippy and can't install it over rustup, cause I should install rust (over rustup) before

I was able to install both rustfmt and clippy just fine via rustup (for stable!) like 1 or 2 days ago via:

`rustup add component {rustfmt, clippy}`

**Comment by whoizit** [comment]
```
$ rustc --version
rustc 1.31.0
$ rustup add component {rustfmt, clippy}
error: Found argument 'add' which wasn't expected, or isn't valid in this context

USAGE:
    rustup [FLAGS] <SUBCOMMAND>

For more information try --help
$ cargo add component {rustfmt, clippy}
Command failed due to unhandled error: Unable to find Cargo.toml


```

**Comment by Cogitri** [comment]
Oops, it's component add not the other way around

**Comment by whoizit** [comment]
```
$ rustup component add {rustfmt, clippy}
error: no default toolchain configured
$ cargo component add {rustfmt, clippy}
error: no such subcommand: `component`
```

**Comment by Cogitri** [comment]
Do as rustup tells you and set a default toolchain: `rustup default stable`

**Comment by whoizit** [comment]
But I want to use rust, clippy, rustfmt from void repos, not from rustup

**Comment by Cogitri** [comment]
...but why?

**Pull request closed without merging**